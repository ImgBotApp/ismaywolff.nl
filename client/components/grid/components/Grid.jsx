import glamorous from 'glamorous'

const setMargin = props => `0 calc(-0.5 * ${props.gutter})`

const Grid = glamorous.div(props => ({
  display: 'flex',
  flexFlow: 'row wrap',
  margin: setMargin(props)
}))

Grid.defaultProps = {
  gutter: 0
}

export default Grid
