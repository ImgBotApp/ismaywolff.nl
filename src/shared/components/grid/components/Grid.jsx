import styled from 'styled-components'

const Grid = styled.div`
  display: flex;
  flex-flow: row wrap;
  margin: 0 calc(-0.5 * ${props => props.gutter});
`

Grid.defaultProps = {
  gutter: 0
}

export default Grid
