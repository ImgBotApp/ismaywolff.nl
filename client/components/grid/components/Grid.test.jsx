import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { matcher, serializer } from 'jest-glamor-react'
import Grid from './Grid'

expect.addSnapshotSerializer(serializer)
expect.extend(matcher)

describe('<Grid />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Grid />)
    expect(shallowToJson(wrapper)).toMatchSnapshotWithGlamor()
  })

  it('renders correctly with gutter', () => {
    const wrapper = shallow(<Grid gutter={'10px'} />)
    expect(shallowToJson(wrapper)).toMatchSnapshotWithGlamor()
  })
})
