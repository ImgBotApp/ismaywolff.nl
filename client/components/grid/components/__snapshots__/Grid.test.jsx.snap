// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> renders correctly 1`] = `
<div
  className="css-13n4620"
/>
`;

exports[`<Grid /> renders correctly with gutter 1`] = `
<div
  className="css-1nd4jt7"
/>
`;
